src/hooks/useDatabase.ts

```diff
@@ -0,0 +1,81 @@
+import { Collection, IApplication, IModelBase, Status } from "~/@types/common";
+
+// taken from https://stackoverflow.com/a/63029283
+type DropFirst<T extends unknown[]> = T extends [any, ...infer U] ? U : never;
+
+const FREE_ID_KEY = 'free-id';
+
+const useDatabase = () => {
+  const get = <T extends IModelBase>(
+    collection: Collection,
+    filterFunction?: (entry: T) => boolean,
+  ): T[] => {
+    const entries: T[] = JSON.parse(localStorage.getItem(collection) || '[]');
+
+    return filterFunction ? entries.filter(filterFunction) : entries;
+  };
+
+  const getById = <T extends IModelBase>(collection: Collection, id: number): T | null => {
+    const entries = get<T>(collection);
+
+    return entries.find((entry) => entry.id === id) || null;
+  };
+
+  const insert = <T extends IModelBase>(
+    collection: Collection,
+    payload: Omit<T, keyof IModelBase>,
+  ): Status => {
+    const freeID = Number(localStorage.getItem(FREE_ID_KEY) || 1);
```

Comment:
replace `||` with `??`

Reply:
I would stay with `||` - it will work in more cases than `??` since `'' || 1 === 1` and `'' ?? 1 === ''`.